backend/lcfs/conftest.py

```diff
@@ -143,7 +143,7 @@ async def dbsession_factory(
 def fastapi_app(
     dbsession: AsyncSession,
     fake_redis_pool: ConnectionPool,
-    set_mock_user_roles,  # Fixture for setting up mock authentication
```

Comment:
I see you renamed this from set_mock_user_roles to set_mock_user, however the function takes roles and not a user. Why do you think this was more clear?

Reply:
In `test_internal_comment.py`, there are a few tests that require a specific Keycloak usernames. I needed to mock these tests with specific user details, for example, in line 367:

```python
set_mock_user(
         fastapi_app,
         ["GOVERNMENT"],
         user_details={
             "keycloak_username": "IDIRUSER"
         }
     )
```

To handle this, I improved `set_mock_user_roles` to support both setting roles and adding user details.